tianshou/policy/modelfree/dqn.py

```diff
@@ -232,11 +233,27 @@ def learn(self, batch: RolloutBatchProtocol, *args: Any, **kwargs: Any) -> TDQNT
 
         return DQNTrainingStats(loss=loss.item())  # type: ignore[return-value]
 
+    @overload
     def exploration_noise(
         self,
-        act: np.ndarray | BatchProtocol,
-        batch: RolloutBatchProtocol,
-    ) -> np.ndarray | BatchProtocol:
+        act: np.ndarray,
+        batch: ObsBatchProtocol,
+    ) -> np.ndarray:
+        pass
+
+    @overload
```

Comment:
@bordeauxred This can be done with generics instead of expressing it with overload, could you adjust it in all relevant places pls?

Reply:
Thanks, resolving! You made the types as class args, not sure if it's a problem. If it is, I might push a minor change. Maybe it does make sense to have a central type instead of repeating the generic, I'll think about it and might adjust